shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java

```diff
@@ -350,6 +354,25 @@ private void anonymizeSOPInstanceUID(int tagInt, Attributes attributes, String m
 		anonymizeTagAccordingToVR(attributes, tagInt, mediaStorageSOPInstanceUID);
 	}
 
+	private void anonymizeFrameOfReferenceUID(int tagInt, Attributes attributes, Map<String, String> frameOfReferenceUIDs) {
+		String value;
+		if (frameOfReferenceUIDs != null && frameOfReferenceUIDs.size() != 0
+				&& frameOfReferenceUIDs.get(attributes.getString(tagInt)) != null) {
+			value = frameOfReferenceUIDs.get(attributes.getString(tagInt));
+		} else {
+			UIDGeneration generator = new UIDGeneration();
+			String newUID = null;
+			try {
+				newUID = generator.getNewUID();
+			} catch (Exception e) {
+				LOG.error(e.getMessage());
+			}
+			value = newUID;
+			frameOfReferenceUIDs.put(attributes.getString(tagInt), value);
+		}
+		anonymizeTagAccordingToVR(attributes, tagInt, value);
+	}
+
 	private void anonymizeSeriesInstanceUID(int tagInt, Attributes attributes, Map<String, String> seriesInstanceUIDs) {
```

Comment:
If I'm not wrong, the three methods could be factorized into a single one (and we just give a different Map<string, string> for each case), but I don't know if it's really relevant in terms of "code clarity" (?)